src/interface.jl

```diff
@@ -2,58 +2,60 @@
 
 
 """
-    isdensitytype(type::Type)
+    hasdensity(d)::Bool
 
-Returns `true` if `type` is compatible with the `DensityInterface` interface,
-otherwise returns `false` by default.
+Returns `true` if `d` is compatible with the `DensityInterface` interface.
+
+`hasdensity(d) == true` implies that `d` is either a density itself or has an
+associated density, e.g. a PDF or a Radon–Nikodym derivative with an implied
+base measure. Also implies that the value of that density at given points can
+be calculated via [`logdensityof`](@ref) and [`densityof`](@ref).
 """
-function isdensitytype end
-export isdensitytype
+function hasdensity end
+export hasdensity
 
-@inline isdensitytype(::Type) = false
+@inline hasdensity(::Any) = false
 
-function check_if_densitytype(type::Type)
-    isdensitytype(type) || throw("Type $(type) is not compatible with DensityInterface")
+function check_hasdensity(d)
+    hasdensity(d) || throw("Object of type $(typeof(d)) is not compatible with DensityInterface")
 end
 
 
 """
-    logdensityof(density, x)::Real
-    logdensityof(density)
+    logdensityof(d, x)::Real
+    logdensityof(d)
 
-Computes the logarithmic value of `density` at a given point `x`, resp.
-returns a function that does so:
+Computes the logarithmic value of density `d` or it's associated density
+at a given point `x`, resp. returns a function that does so:
```

Comment:
Maybe it would be clearer to add two separate docstrings, ie one for the regular and one for the curried version.

Reply:
Docstring are split now.